source/mqtt_agent.c

```diff
@@ -217,6 +217,30 @@ static MQTTStatus_t resendPublishes( MQTTAgentContext_t * pMqttAgentContext );
  */
```

Comment:
it would be nicer if you split the tests from the code
1- the commit title would be more relevant - easier to find commits and what they actually did
2- if you had to revert the impact would be smaller and re-merging easier
3- single responsibility of commits
..